packages/ra-ui-materialui/src/field/ReferenceManyCount.tsx

```diff
@@ -69,22 +74,21 @@ export const ReferenceManyCount = (props: ReferenceManyCountProps) => {
     );
 
     return link ? (
-        // @ts-ignore TypeScript complains that the props for <a> aren't the same as for <span>
-        <Link
-            to={{
-                pathname: createPath({ resource: reference, type: 'list' }),
-                search: `filter=${JSON.stringify({
-                    ...(filter || {}),
-                    [target]: record[source],
-                })}`,
-            }}
-            variant="body2"
-            component="span"
-            onClick={e => e.stopPropagation()}
-            {...rest}
-        >
-            {body}
-        </Link>
+        <Box component="span" {...rest}>
```

Comment:
We don't need a wrapping span in that case, plus it's a breaking change since the rest parameters no longer apply to the Link.

Reply:
Right. In this case, it seems to me that https://github.com/marmelab/react-admin/pull/8752 was the better solution then! I'll reopen it.